contracts/Comptroller.sol

```diff
@@ -1021,6 +1036,62 @@ contract Comptroller is ComptrollerV3Storage, ComptrollerInterface, ComptrollerE
         allMarkets.push(CToken(cToken));
     }
 
+
+    /**
+      * @notice Set the given borrow limit for the given cToken market
+      * @dev Admin or borrowLimitGuardian function to set the borrow limit
+      * @param cToken The address of the market (token) to change the borrow limit for
+      * @param borrowLimit The new borrow limit value in underlying (maximum borrowing) to be set
+      * @return uint 0=success, otherwise a failure. (See enum Error for details)
+      */
+    function _setMarketBorrowLimit(CToken cToken, uint256 borrowLimit) external {
+        // Allow borrow limit guardian to change borrow limit too.
+        require(msg.sender == admin || msg.sender == borrowLimitGuardian, "Only admin or Borrow Limit Guardian can set market borrow limits"); 
+
+        _setMarketBorrowLimitInternal(cToken,borrowLimit);
+    }
+
+    function _setMarketBorrowLimits(CToken[] calldata cTokens, uint256[] calldata newBorrowLimits) external {
+    	require(msg.sender == admin || msg.sender == borrowLimitGuardian, "Only admin or Borrow Limit Guardian can set market borrow limits"); 
+
+        uint numMarkets = cTokens.length;
+        uint numLimits = newBorrowLimits.length;
+
+        require(numMarkets != 0 && numMarkets == numLimits, "Invalid input");
+
+        for(uint8 i = 0; i < numMarkets; i++) {
+        	_setMarketBorrowLimitInternal(cTokens[i],newBorrowLimits[i]);
+        }
+    }
+
+    function _setMarketBorrowLimitInternal(CToken cToken, uint256 borrowLimit) internal {
+    	borrowLimits[address(cToken)] = borrowLimit;
+        emit NewBorrowLimit(cToken, borrowLimit);
+    }
+
+
+    function getMarketBorrowLimit(CToken cToken) public view returns (uint) {
+        return borrowLimits[address(cToken)];
```

Comment:
this storage variable is public, right? is the getter necessary?

Reply:
Correct. Completely unneeded, and we are approaching max contract size too, so this really shouldn't be here. Thanks